### PR TITLE
Fixing package vulnerabilities for y18n and netmask

### LINKED ISSUE #10764: y18n, netmask dependency vulnerabilities
Output of `.circleci/scripts/yarn-audit.sh`:

```
yarn audit v1.22.5
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ high          │ Prototype Pollution                                          │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ y18n                                                         │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=5.0.5||>=4.0.1 <5.0.0||>=3.2.2 <4.0.0                      │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ 3box                                                         │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ 3box > ipfs > cid-tool > yargs > y18n                        │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://www.npmjs.com/advisories/1654                        │
└───────────────┴──────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ high          │ Prototype Pollution                                          │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ y18n                                                         │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=5.0.5||>=4.0.1 <5.0.0||>=3.2.2 <4.0.0                      │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ 3box                                                         │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ 3box > ipfs > yargs > y18n                                   │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://www.npmjs.com/advisories/1654                        │
└───────────────┴──────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ high          │ netmask npm package vulnerable to octal input data           │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ netmask                                                      │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=2.0.1                                                      │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ pubnub                                                       │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ pubnub > superagent-proxy > proxy-agent > pac-proxy-agent >  │
│               │ pac-resolver > netmask                                       │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://www.npmjs.com/advisories/1658                        │
└───────────────┴──────────────────────────────────────────────────────────────┘
9 vulnerabilities found - Packages audited: 1623
Severity: 6 Low | 3 High
Done in 2.29s.
Audit shows 3 moderate or high severity advisories _in the production dependencies_

Exited with code exit status 1
CircleCI received exit code 1
```

### PR DESCRIPTION
**Pull request by ryanml**
Manually resolving netmask to ^2.0.1, and y18n to ^4.0.1

Fixes: https://github.com/MetaMask/metamask-extension/issues/10764

### CODE REVIEW

**Review by DONALDSULLENDER — DISMISSED** [review]

File: yarn.lock [hunk 1]
```diff
@@ -25753,20 +25753,10 @@ xtend@~3.0.0:
...
-  integrity sha512-r9S/ZyXu/Xu9q1tYlpsLIsa3EeLXXk0VwlxqTcFRfg9EhMW+17kbt9G0NrgCmhGb5vT2hyhJZLfDGx+7+5Uj/w==
-
-y18n@^5.0.5:
-  version "5.0.5"
-  resolved "https://registry.yarnpkg.com/y18n/-/y18n-5.0.5.tgz#8769ec08d03b1ea2df2500acef561743bbb9ab18"
-  integrity sha512-hsRUr4FFrvhhRH12wOdfs38Gy7k2FFzB9qgN9v3aLykRq0dRcdcpz5C9FxdS2NuhOrI/628b/KSTJ3rwHysYSg==
+y18n@^3.2.1, y18n@^4.0.0, y18n@^4.0.1, y18n@^5.0.5:
+  version "4.0.1"
+  resolved "https://registry.yarnpkg.com/y18n/-/y18n-4.0.1.tgz#8db2b83c31c5d75099bb890b23f3094891e247d4"
+  integrity sha512-wNcy4NvjMYL8gogWWYAO7ZFWFfHcbdbE57tZO8e4cbpj8tfUcwrwqSl3ad8HxpYWCdXcJUCeKKZS62Av1affwQ==
```
Comment (danjm) [inline yarn.lock:25759]: Should "Selective version resolution", to specify versions of specific package sub-dependencies, be used here? https://classic.yarnpkg.com/en/docs/selective-version-resolutions/ (As is done, for example, with 3box and node-forge)

We have many dependencies that rely on y18n, and I think this forces all of them to use v4.0.1, which could be breaking for any that rely on v3.2.1 or v5.0.5

Comment (ryanml) [reply yarn.lock:25759]: I think you're absolutely right here, though I did some manual testing, something that should be taken in to consideration. I'll make the update with the selective version resolution

**Review by Gudahtt — CHANGES_REQUESTED** [review]
Yarn resolutions are only necessary when we need to make an update that's not compatible with the range requested by the dependency. In this case, for `y18n`, all we need to do is update the lockfile to use `4.0.1`, since the range asked for is `^4.0.0` which includes `4.0.1`

**Review by Az111z — APPROVED** [review]

**Review by Az111z — DISMISSED** [review]
A

**Comment by ryanml** [comment]
changes pushed

**Review by Gudahtt — APPROVED** [review]
LGTM!

File: yarn.lock (hunk 1 above)
Comment (ryanml) [reply yarn.lock:25759]: newest changes target specific, vulnerable versions of `y18n`, and it no longer forces a version on particular resolutions

**Pull request merged**